auth/password_authenticator.cc

```diff
@@ -106,13 +116,17 @@ future<> password_authenticator::migrate_legacy_metadata() const {
 }
 
 future<> password_authenticator::create_default_if_missing() const {
-    return default_role_row_satisfies(_qp, &has_salted_hash).then([this](bool exists) {
+    return default_role_row_satisfies(_qp, &has_salted_hash, _superuser).then([this](bool exists) {
         if (!exists) {
+            std::string salted_pwd(get_config_value(_qp.db().get_config().auth_superuser_salted_password(), ""));
+            if (salted_pwd.empty()) {
+                salted_pwd = passwords::hash(DEFAULT_USER_PASSWORD, rng_for_salt);
+            }
```

Comment:
This is frightening, esp. in relation with unfixed #1796. After a topology change and before a repair, this acts as a password reset to a known password, no? Though I see the patch doesn't introduce the problem.

Reply:
This only sets a default IFF there is no entry in auth tables already. If you have already initialized the system with either option of just a normal "create user..." thingy, it will do nothing. As stated in the config.cc section.

Reply:
Lack of #1796 means that when adding a datacenter or when increasing the replication factor, there's a window (before repair) where queries return nothing.